lib/shipit/same_site_cookie_middleware.rb

```diff
@@ -0,0 +1,30 @@
+module Shipit
+  class SameSiteCookieMiddleware
+    COOKIE_SEPARATOR = "\n".freeze
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      status, headers, body = @app.call(env)
+
+      if headers && headers['Set-Cookie'] &&
+         Rack::Request.new(env).ssl?
+
+        set_cookies = headers['Set-Cookie']
```

Comment:
Something to check here. What if the response has multiple `Set-Cookie` headers? AFAIK it's possible right?

Reply:
Looking at rack's implementation, seems like it can indeed be an array: https://github.com/rack/rack/blob/f9ad97fd69a6b3616d0a99e6bedcfb9de2f81f6c/lib/rack/utils.rb#L219-L258